modules/mobile/app/controllers/mobile/v0/appointments_controller.rb

```diff
@@ -37,16 +37,39 @@ def index
       end
 
       def cancel
-        decoded_cancel_params = Mobile::V0::Appointment.decode_cancel_id(params[:id])
-        contract = Mobile::V0::Contracts::CancelAppointment.new.call(decoded_cancel_params)
-        raise Mobile::V0::Exceptions::ValidationErrors, contract if contract.failure?
+        id = params[:id]
+        # appointment request cancel ids is appointment id while other appointments will be encoded string
+        if uuid?(id)
+          appointment_request = appointments_proxy.get_appointment_request(id)
+          raise Common::Exceptions::BackendServiceException, 'MOBL_502_upstream_error' if appointment_request.nil?
+
+          appointment_status = appointment_request['status']
+          if appointment_status == 'Submitted'
```

Comment:
Technically the VAOs endpoint will accept already cancelled requests and just give back the same thing but wanted to avoid hiding the issue.

Reply:
That's fair. Probably want to mirror the logic for cancelling confirmed appointments which doesn't do any checks on status before attempting to cancel. I'll confirm with FE but will assume for now they will want it consistent and update that.

Reply:
I'm not sure that there is a clear right or wrong here. Logging an error if it's already been cancelled could reveal issues, but it will also almost surely sometimes be the cause of a race condition and cause us to think there's an issue when there isn't. Not checking first has multiple advantages though: the request will be faster, the code will be simpler, and we don't have to mess around another VCR cassette in specs. And my question about 404s becomes irrelevant.

Reply:
Thats a good argument to pass return the 200 without any additional error logging. I've updated it to do that.